integrationTests/chainSimulator/staking/jail_test.go

```diff
@@ -0,0 +1,145 @@
+package staking
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data/transaction"
+	"github.com/multiversx/mx-chain-go/config"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/components/api"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/configs"
+	"github.com/multiversx/mx-chain-go/vm"
+	"github.com/stretchr/testify/require"
+)
+
+// Test description
+// All test cases will do a stake transaction and wait till the new node is jailed
+// testcase1 -- unJail transaction will be sent when staking v3.5 is still action --> node status should be `new` after unjail
+// testcase2 -- unJail transaction will be sent when staking v4 step1 is action --> node status should be `auction` after unjail
+// testcase3 -- unJail transaction will be sent when staking v4 step2 is action --> node status should be `auction` after unjail
+// testcase4 -- unJail transaction will be sent when staking v4 step3 is action --> node status should be `auction` after unjail
+func TestChainSimulator_ValidatorJailUnJail(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	t.Run("staking ph 4 is not active", func(t *testing.T) {
+		testChainSimulatorJailAndUnJail(t, 4, "new")
+	})
+
+	t.Run("staking ph 4 step 1 active", func(t *testing.T) {
+		testChainSimulatorJailAndUnJail(t, 5, "auction")
+	})
+
+	t.Run("staking ph 4 step 2 active", func(t *testing.T) {
+		testChainSimulatorJailAndUnJail(t, 6, "auction")
+	})
+
+	t.Run("staking ph 4 step 3 active", func(t *testing.T) {
+		testChainSimulatorJailAndUnJail(t, 7, "auction")
+	})
+}
+
+func testChainSimulatorJailAndUnJail(t *testing.T, targetEpoch int32, nodeStatusAfterUnJail string) {
+	startTime := time.Now().Unix()
+	roundDurationInMillis := uint64(6000)
+	roundsPerEpoch := core.OptionalUint64{
+		HasValue: true,
+		Value:    20,
+	}
+
+	numOfShards := uint32(3)
+
+	cs, err := chainSimulator.NewChainSimulator(chainSimulator.ArgsChainSimulator{
+		BypassTxSignatureCheck:   false,
+		TempDir:                  t.TempDir(),
+		PathToInitialConfig:      defaultPathToInitialConfig,
+		NumOfShards:              numOfShards,
+		GenesisTimestamp:         startTime,
+		RoundDurationInMillis:    roundDurationInMillis,
+		RoundsPerEpoch:           roundsPerEpoch,
+		ApiInterface:             api.NewNoApiInterface(),
+		MinNodesPerShard:         3,
+		MetaChainMinNodes:        3,
+		NumNodesWaitingListMeta:  1,
+		NumNodesWaitingListShard: 1,
+		AlterConfigsFunction: func(cfg *config.Configs) {
+			cfg.EpochConfig.EnableEpochs.StakingV4Step1EnableEpoch = 5
```

Comment:
Can we add some consts for these stakingV4 steps so that they are easily "linked" between these values and the ones you use when you call `testChainSimulatorJailAndUnJail` ?

Reply:
added